### PR TITLE
network: cleanups for uplink interface handling for RADV and DHCP6-PD

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/network/networkd-dhcp6.c [hunk 1]
```diff
@@ -42,25 +42,28 @@ bool link_dhcp6_pd_is_enabled(Link *link) {
...
 }
 
-static int dhcp6_pd_resolve_uplink(Link *link, Link **ret) {
+static bool dhcp6_pd_is_uplink(Link *link, Link *target, bool accept_auto) {
+        assert(link);
+        assert(target);
+
+        if (!link_dhcp6_pd_is_enabled(link))
+                return false;
+
```
Comment (bluca) [inline src/network/networkd-dhcp6.c:51]: `assert(link->network)` or check

File: src/network/networkd-dhcp6.c [hunk 2]
```diff
@@ -1456,27 +1446,37 @@ static bool dhcp6_pd_uplink_is_ready(Link *link) {
...
 
         assert(link);
         assert(link->manager);
-        assert(link->network);
+        assert(link_dhcp6_pd_is_enabled(link));
         assert(ret);
 
-        if (dhcp6_pd_resolve_uplink(link, &l) >= 0) {
-                if (!dhcp6_pd_uplink_is_ready(l))
+        if (link->network->dhcp6_pd_uplink_name)
```
Comment (bluca) [inline src/network/networkd-dhcp6.c:1450]: `assert(link->network)` or check

**Review by bluca — APPROVED** [review]
just a couple of nits, lgtm

File: src/network/networkd-dhcp6.c (hunk 1 above)
Comment (yuwata) [reply src/network/networkd-dhcp6.c:51]: `link_dhcp6_pd_is_enabled()` already checks that.

File: src/network/networkd-dhcp6.c (hunk 2 above)
Comment (yuwata) [reply src/network/networkd-dhcp6.c:1450]: Also here. That is already checked by `link_dhcp6_pd_is_enabled()` in the above.

**Pull request merged**